File: receiver.py
from aes_ops import decrypt_cbc

def main():
    key_hex = input("Enter key (hex): ").strip()
    iv_hex = input("Enter IV (hex): ").strip()
    ct_hex = input("Enter ciphertext (hex): ").strip()

    key = bytes.fromhex(key_hex)
    iv = bytes.fromhex(iv_hex)
    ciphertext = bytes.fromhex(ct_hex)

    try:
        plaintext = decrypt_cbc(ciphertext, key, iv)
        print("\nDecrypted text:")
        print(plaintext.decode("utf-8"))
    except Exception as e:
        print(f"\nDecryption failed (wrong key or error).\n{e}")

if __name__ == "__main__":
    main()

File: src_mat/receiver.m
key_hex = strtrim(input('Enter key (hex): ', 's'));
iv_hex  = strtrim(input('Enter IV (hex): ', 's'));
ct_hex  = strtrim(input('Enter ciphertext (hex): ', 's'));

try
    key        = hex2bytes(key_hex);
    iv         = hex2bytes(iv_hex);
    ciphertext = hex2bytes(ct_hex);

    plaintext = decrypt_cbc(ciphertext, key, iv);

    fprintf('\nDecrypted text:\n');
    fprintf('%s\n', char(plaintext));
catch ME
    fprintf('\nDecryption failed (wrong key or error).\n%s\n', ME.message);
end

% ---- helper local function ----
function b = hex2bytes(h)
    h = regexprep(h, '\s', '');   % bỏ khoảng trắng nếu có
    if mod(length(h), 2) ~= 0
        error('Hex string length must be even.');
    end
    b = uint8(sscanf(h, '%2x').');  % mỗi 2 ký tự hex -> 1 byte
end
